cardano-config/src/Cardano/Config/Shelley/KES.hs

```diff
@@ -29,43 +33,62 @@ genKESKeyPair duration = do
   let verKeyKes = deriveVerKeyKES signKeyKES
   pure $ (VKeyES verKeyKes,SKeyES signKeyKES)
 
-data KESError = ReadKESSigningKeyError !FilePath !IOException
+data KESError = ReadKESSigningKeyError !TextViewFileError
               | ReadKESVerKeyError !FilePath !IOException
-              | DecodeKESSigningKeyError !FilePath !CBOR.DecoderError
               | DecodeKESVerKeyError !FilePath !CBOR.DecoderError
-              | WriteKESSigningKeyError !FilePath !IOException
+              | WriteKESSigningKeyError !TextViewFileError
               | WriteKESVerKeyError !FilePath !IOException
 
-readKESSigningKey :: FilePath ->  ExceptT KESError IO (SKeyES TPraosStandardCrypto)
-readKESSigningKey fp = do
-  bs <- handleIOExceptT (ReadKESSigningKeyError fp) $ LB.readFile fp
-  firstExceptT (DecodeKESSigningKeyError fp) . hoistEither $ second SKeyES $ CBOR.decodeFull bs
-
-renderKESError :: KESError -> String
+renderKESError :: KESError -> Text
 renderKESError kesErr =
   case kesErr of
-    ReadKESSigningKeyError fp ioExcptn -> "KES signing key read error at: " <> fp
-                                          <> " Error: " <> show ioExcptn
+    ReadKESSigningKeyError err -> "KES signing key read error: " <> renderTextViewFileError err
+
+    ReadKESVerKeyError fp ioExcptn -> "KES verification key read error at: " <> toS fp
+                                      <> " Error: " <> textShow ioExcptn
+
+
+    DecodeKESVerKeyError fp cborDecErr -> "KES verification key decode error at: " <> toS fp
+                                          <> " Error: " <> textShow cborDecErr
+
+    WriteKESSigningKeyError err -> "KES signing key write error: " <> renderTextViewFileError err
 
-    ReadKESVerKeyError fp ioExcptn -> "KES verification key read error at: " <> fp
-                                      <> " Error: " <> show ioExcptn
+    WriteKESVerKeyError fp ioExcptn -> "KES verification key write error at: " <> toS fp
+                                       <> " Error: " <> textShow ioExcptn
 
-    DecodeKESSigningKeyError fp cborDecErr -> "KES signing key decode error at: " <> fp
-                                              <> " Error: " <> show cborDecErr
+encodeKESSigningKey :: SKeyES TPraosStandardCrypto -> TextView
+encodeKESSigningKey (SKeyES sKeyEs) =
+  encodeToTextView tvType' tvTitle' CBOR.toCBOR sKeyEs
+ where
+  tvType' = TextViewType "SKeyES TPraosStandardCrypto"
+  tvTitle' = TextViewTitle "KES Signing Key"
```

Comment:
These have `IsString` instances, so we don't need to provide the newtype wrapper, and we could just do it inline:
```suggestion
    encodeToTextView
      "SKeyES TPraosStandardCrypto"
      "KES Signing Key"
      CBOR.toCBOR
      sKeyEs
```
We should also pick more meaningful external type names. It does not need to correspond to our internal type names like "SKeyES TPraosStandardCrypto".

Reply:
Ah missed this one!